core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -459,7 +469,7 @@ private static AnnotatedValueResolver of(AnnotatedElement annotatedElement,
 
         final Header header = annotatedElement.getAnnotation(Header.class);
         if (header != null) {
-            final String name = findName(header, typeElement);
+            final String name = findName(typeElement, header.value(), v -> toHeaderName(v));
```

Comment:
It is difficult to understand exactly what `delegate` is used for. If the function normalizes the result, `delegate` would not be a proper name.

As `findName(Object nameRetrievalTarget, String value, Function<String, String> delegate)` is only used here, what do you think of applying `toHeaderName()` to the returned value?
```suggestion
            final String name = toHeaderName(findName(typeElement, header.value());
```

Reply:
@ikhoon nim, thank you for your comment! 🙇‍♂️🙇‍♂️ 
I agree with you! 

I have a question. If I accept your commit suggestion, there's no need to keep the function below:
`static String findName(Object nameRetrievalTarget, String value, Function<String, String> delegate) {...}`. This is because only the `static String findName(Object nameRetrievalTarget, String value) {...}` method is in use.

Should I remove the method static String findName(Object nameRetrievalTarget, String value, Function<String, String> delegate) {...}, or is it okay to leave it?

`Armeria` is a rapidly evolving framework, and it's possible that that methods might be used in the near future, so it might be a good idea to leave it as it is. What do you think?

Reply:
> Should I remove the method static String findName(Object nameRetrievalTarget, String value, Function<String, String> delegate) {...}, 

I prefer to remove `findName(Object nameRetrievalTarget, String value, Function<String, String> delegate)`.

> and it's possible that that methods might be used in the near future, so it might be a good idea to leave it as it is.

It is a simple logic. We can easily add it if we need to generalize the function. Reference: https://www.makeuseof.com/tag/basic-programming-principles/#you-aren-39-t-going-to-need-it-yagni
Although we don't have to follow it, I have a similar principle in mind.

Reply:
I see. it's good guild line. I'll keep in mind as well.  
Anyway, i removed `findName(Object nameRetrievalTarget, String value, Function<String, String> delegate).`